### PR TITLE
Nova Scotia Pension Income Amount

### PR DESCRIPTION
**Pull request by okeyiii**
#473 

Add pension amount for Nova Scotia

### CODE REVIEW

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/pension_income_amount/max_amount.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  2022-01-01: 1_173
+metadata:
+  unit: currency-CAD
+  label: Nova Scotia pension income amount max amount
+  reference: 
+  - title: 2022 Nova Scotia Personal Tax Credits Return
+    href: https://hr.acadiau.ca/files/sites/hr/Payroll/Pensions%20&%20Benefits/NS_TD1_2022.pdf#page=1
+  - title: 2022 Worksheet NS428
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5003-c/5003-c-22e.pdf#page=1
+  - title: Nova Scotia income tax act - subdivision c - Deduction for employment out of Canada
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/pension_income_amount/max_amount.yaml:12]: more precise reference point

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/pension_income_amount/max_amount.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+description: Nova Scotia max amount for pension income amount.
+values: 
+  2022-01-01: 1_173
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/pension_income_amount/max_amount.yaml:3]:
```suggestion
  2017-01-01: 1_173
```

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/pension_income_amount/max_amount.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  unit: currency-CAD
+  label: Nova Scotia pension income amount max amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/pension_income_amount/max_amount.yaml:6]:
```suggestion
  label: Nova Scotia pension income amount cap
```

File: policyengine_canada/tests/gov/provinces/ns/tax/income/credits/ns_pension_income_amount
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/ns/tax/income/credits/ns_pension_income_amount:1]: add .yaml

File: policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_pension_income_amount
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_pension_income_amount:1]: add .py

File: policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_pension_income_amount [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from policyengine_canada.model_api import *
+
+
+class ns_pension_income_amount(Variable):
+    value_type = float
+    entity = Person
+    label = "Nova Scotia Pension Income Amount"
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_pension_income_amount:7]:
```suggestion
    label = "Nova Scotia pension income amount"
```

File: policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_pension_income_amount [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        "https://nslegislature.ca/sites/default/files/legc/statutes/income%20tax.pdf#page=28",
+    )
+
+    def formula(person, period, parameters):
+        max_amount = parameters(
+            period
+        ).gov.provinces.ns.tax.income.credits.pension_income_amount.max_amount
+        pension_income_amount = person("pension_and_savings_plan_income", period)
+
+        return min(pension_income_amount, max_amount)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_pension_income_amount:23]:
```suggestion
        return min_(pension_income_amount, max_amount)
```

File: policyengine_canada/tests/gov/provinces/ns/tax/income/credits/ns_pension_income_amount
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/ns/tax/income/credits/ns_pension_income_amount:1]: add one more test with 0

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_pension_income_amount.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    definition_period = YEAR
+    defined_for = ProvinceCode.NS
+    reference = (
+        "https://hr.acadiau.ca/files/sites/hr/Payroll/Pensions%20&%20Benefits/NS_TD1_2022.pdf#page=1",
+        "https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5003-c/5003-c-22e.pdf#page=1",
+        "https://nslegislature.ca/sites/default/files/legc/statutes/income%20tax.pdf#page=28",
+    )
+
+    def formula(person, period, parameters):
+        cap = parameters(
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_pension_income_amount.py:18]: our convention, then `p.cap`
```suggestion
        p = parameters(
            period
        ).gov.provinces.ns.tax.income.credits.pension_income_amount
```

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/pension_income_amount/cap.yaml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+description: Nova Scotia caps the pension income amount at this amount. 
+values: 
+  2017-01-01: 1_173
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Nova Scotia pension income amount cap 
+  reference: 
+  - title: 2022 Nova Scotia Personal Tax Credits Return
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/pension_income_amount/cap.yaml:9]:
```suggestion
  - title: 2022 Nova Scotia Personal Tax Credits Return Line 3
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Can you also hook it up to the net income tree?

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: link to the issue in the PR description

File: policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_pension_income_amount.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+            period
+        ).gov.provinces.ns.tax.income.credits.pension_income_amount
+
+        cap = p.cap
+
+        pension_income_amount = person(
+            "pension_and_savings_plan_income", period
+        )
+
+        return min_(pension_income_amount, cap)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_pension_income_amount.py:28]:
```suggestion
        return min_(pension_income_amount, p.cap)
```

File: policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_pension_income_amount.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        "https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5003-c/5003-c-22e.pdf#page=1",
+        "https://nslegislature.ca/sites/default/files/legc/statutes/income%20tax.pdf#page=28",
+    )
+
+    def formula(person, period, parameters):
+        p = parameters(
+            period
+        ).gov.provinces.ns.tax.income.credits.pension_income_amount
+
+        cap = p.cap
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_pension_income_amount.py:22]:
```suggestion
```

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/pension_income_amount/cap.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/pension_income_amount/cap.yaml:1]: is there an age restriction for this credit?

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/pension_income_amount/cap.yaml
Comment (okeyiii) [reply policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/pension_income_amount/cap.yaml:1]: https://www.nspssp.ca/members/your-working-years/pension-benefit/how-your-pension-calculated

**Comment by PavelMakarchuk** [comment]
> again please hook up to net income tree

@MaxGhenis we will create a new PR to create the net income tree of NS once the open PR's are merged, since we have not followed this practice early in the development the credits are scattered in the repo. 
@okeyiii and I thought this would be the best approach to have a clean PR for province income tree structures that can then be used as a reference for other devs.

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**